wyrm_nomcu.py

```diff
@@ -215,598 +93,76 @@ def __init__(self, revision, sys_clk_freq=50e6, toolchain="trellis",
+            platform.add_extension([
+                ("rgb_output", connector,
+                    Subsignal("panel_r0", Pins(f"j{connector}:{r_offset}")),
+                    Subsignal("panel_g0", Pins(f"j{connector}:{g_offset}")),
+                    Subsignal("panel_b0", Pins(f"j{connector}:{b_offset}")),
+                    Subsignal("panel_r1", Pins(f"j{connector}:{r_offset+4}")),
+                    Subsignal("panel_g1", Pins(f"j{connector}:{g_offset+4}")),
+                    Subsignal("panel_b1", Pins(f"j{connector}:{b_offset+4}")),
+                    IOStandard("LVCMOS33"))])
 
-        j8r0 = platform.request("panel_r0",    8);
-        j8g0 = platform.request("panel_g0",    8);
-        j8b0 = platform.request("panel_b0",    8);
-        j8r1 = platform.request("panel_r1",    8);
-        j8g1 = platform.request("panel_g1",    8);
-        j8b1 = platform.request("panel_b1",    8);
-        #j8E = platform.request("panel_e",     8);
-        #j8A = platform.request("panel_a",     8);
-        #j8B = platform.request("panel_b",     8);
-        #j8C = platform.request("panel_c",     8);
-        #j8D = platform.request("panel_d",     8);
-        #j8clk = platform.request("panel_clk", 8);
-        #j8stb = platform.request("panel_stb", 8);
-        #j8oe = platform.request("panel_oe",   8);
-
-        self.comb += j8r0.eq(s_j8r0)
-        self.comb += j8g0.eq(s_j8g0)
-        self.comb += j8b0.eq(s_j8b0)
-        self.comb += j8r1.eq(s_j8r1)
-        self.comb += j8g1.eq(s_j8g1)
-        self.comb += j8b1.eq(s_j8b1)
-        #self.comb += j8A.eq(s_j8a)
-        #self.comb += j8B.eq(s_j8b)
-        #self.comb += j8C.eq(s_j8c)
-        #self.comb += j8D.eq(s_j8d)
-        #self.comb += j8E.eq(s_j8e)
-        #self.comb += j8clk.eq(s_j8clk)
-        #self.comb += j8stb.eq(s_j8stb)
-        #self.comb += j8oe.eq(s_j8oe)
-        self.comb += s_j8_ctrl_en.eq(s_shared_en[4])
-        self.comb += s_j8_ctrl_addr.eq(s_shared_addr)
-        self.comb += s_j8_ctrl_wdat.eq(s_shared_wdat)
+        # LED Panel --------------------------------------------------------------------------------
+        self.ctrl_signals = ctrl_signals = Record([
+            ("en", 8),
+            ("addr", 16),
+            ("wdat", 24),
+        ])
+
+        s_shared_en = self.ctrl_signals.en
+        s_shared_addr = self.ctrl_signals.addr
+        s_shared_wdat = self.ctrl_signals.wdat
+
+        self.add_ledpanel(connector=4, select_line=0,
+            shared_output=self.platform.request('shared_output'))
+        self.add_ledpanel(connector=3, select_line=1)
+        self.add_ledpanel(connector=2, select_line=2)
+        self.add_ledpanel(connector=1, select_line=3)
+        self.add_ledpanel(connector=5, select_line=4)
+        self.add_ledpanel(connector=6, select_line=4) # review select_line=5?
+        self.add_ledpanel(connector=7, select_line=4) # review select_line=6?
+        self.add_ledpanel(connector=8, select_line=4) # review select_line=7?
```

Comment:
when you fix the select_line=4 issue, I would also just make the `select_line` and `connector` numbers match - there's no reason to have them swapped except the order I was testing them in on my desk